front/src/applications/operationalStudies/components/MacroEditor/osrdToNge.ts

```diff
@@ -419,22 +474,25 @@ const getNgeLabels = (state: MacroEditorState): LabelDto[] => [
 ];
 
 /**
- * Return a compatible object for NGE
+ * Return a compatible object for NGE.
  */
-export const getNgeDto = (state: MacroEditorState): NetzgrafikDto => {
+export const getNgeDto = (
+  state: MacroEditorState,
+  t: (key: string, options?: Record<string, unknown>) => string
+): NetzgrafikDto => {
   const labels = getNgeLabels(state);
```

Comment:
It would be better to create the trainrunFrequencies here
```suggestion
  const labels = getNgeLabels(state);
  const trainrunFrequencies = getNgeFrequencies(state, t);
```

The function `getNgeFrequencies` should:
- get the default frequencies
- add the new frequencies if needed
- sort and re-order the frequencies
- return the frequencies

Reply:
I'm not sure to understand the full workflow of the two calls, and how the `trainrunFrequencies` are dynamically shared between the two functions.

If I understand right, `trainrunFrequencies` is already populated at the first call of `loadAndIndexNge()`, then we have only to get this list in `getNgeDto()`.

Is there somehting I am missing?

Reply:
No sorry. What I meant was that actually, the frequencies need to be stored in the MacroEditorState, and you should compute them in `loadAndIndexNge()`
You can do something like this (in `loadAndIndexNge()`):
```
state.trainrunFrequencies = getNgeFrequencies(state.timetableItems, t);
```
The `getNgeFrequencies` will replace your functions `initializeTrainrunFrequencies` and `createTrainrunFrequencyFromPace`.
Once you have all the frequencies in `state.trainrunFrequencies`, your function `getOrCreateTrainrunFrequency` can be renamed `getTrainrunFrequency` and simplified a bit :)

Reply:
Done in a specific comment "refactor trainrun frequencies". It was indeed much clearer that way!